README.md

```diff
@@ -32,6 +32,9 @@ pub struct TestCircuit {
 }
 
 impl Circuit for TestCircuit {
+    // You could generate the ID by hashing the `gadget()` fn code
+    // in a proc_macro with something like Blake3.
+    const CIRCUIT_ID: [u8; 32] = *b"0x000000000000000000000000000000";
```

Comment:
That is a bit misleading. Since the string starts with `0x` looks like a hex representation of a number of 30 bytes when instead is just a string of 32 bytes (could be `b"xx000000000000000000000000000000"` (and was still valid).

For the test, I would just use an array filled with the same value:
`const CIRCUIT_ID: [u8; 32] = [0xff; 32];`

It's less misleading.